packages/base-driver/lib/protocol/protocol.js

```diff
@@ -454,7 +454,16 @@ function buildHandler(app, method, path, spec, driver, isSessCmd) {
     } catch (err) {
       // if anything goes wrong, figure out what our response should be
       // based on the type of error that we encountered
-      let actualErr = err;
+      let actualErr;
+      if (!(err instanceof Error) || !(_.has(err, 'stack') && _.has(err, 'message'))) {
+        getLogger(driver, req.params.sessionId || newSessionId).warn(
+          'The thrown error object does not seem to be a valid instance of the Error class. This ' +
+          'might be a genuine bug of a driver or a plugin.'
+        );
+        actualErr = new Error(`${err ?? 'unknown'}`);
+      } else {
+        actualErr = err;
+      }
```

Comment:
If the symptom is just that we cannot use the `in` operator on something thrown, then all we _really_ need to do is check if `err` is an object:

```js
const actualErr = _.isObject(err) ? err : new Error(`${err ?? 'unknown'}`);
```

right?  because the only thing you can't use `in` on (to my knowledge) are primitives and `null`, both of which `_.isObject()` returns `false` for.

I'm also unsure if _this_ is the only place we need to be making these sorts of checks.  If we cannot make guarantees about the behavior of drivers, we _may_ need to make these coercions happen wherever `BaseDriver` (or `AppiumDriver`, which I think calls any `updateServer` methods?) explicitly catches something thrown by an extension.  This goes for `Promise`s as well, since there's no requirement for a `Promise` to reject with anything in particular.

Unless you think you want to take a look at where else this might be a problem, I'd say let's just get this PR merged and deal with the rest later.  There may be a better way to trap anything coming out of extensions via async hooks or something.

Reply:
javascript is weird and null is an object (while undefined isn't). 
Also we don't want to accept any object, but an error, where we have a proper stacktrace attached.